source/code/plugin/filter_inventory2mdm.rb

```diff
@@ -0,0 +1,261 @@
+        @@node_count_metric_name = 'nodesCount'
+        @@pod_count_metric_name = 'podCount'
+        @@pod_inventory_tag = 'mdm.kubepodinventory'
+        @@node_inventory_tag = 'mdm.kubenodeinventory'
+        @@node_status_ready = 'Ready'
+        @@node_status_not_ready = 'NotReady'
+
+        @@node_inventory_custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "insights.container/nodes", 
+                        "dimNames": [ 
+                        "status"
+                        ], 
+                        "series": [ 
+                        { 
+                            "dimValues": [ 
+                            "%{statusValue}"
+                            ], 
+                            "min": %{node_status_count},
+                            "max": %{node_status_count}, 
+                            "sum": %{node_status_count}, 
+                            "count": 1
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+        @@pod_inventory_custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "insights.container/nodes", 
+                        "dimNames": [ 
+                        "phase", 
+                        "namespace", 
+                        "node", 
+                        "controllerName"
+                        ], 
+                        "series": [ 
+                        { 
+                            "dimValues": [ 
+                            "%{phaseDimValue}", 
+                            "%{namespaceDimValue}", 
+                            "%{nodeDimValue}", 
+                            "%{controllerNameDimValue}"
+                            ], 
+                            "min": %{podCountMetricValue},
+                            "max": %{podCountMetricValue}, 
+                            "sum": %{podCountMetricValue}, 
+                            "count": 1 
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+        @process_incoming_stream = true
+
+		def initialize
+            super
+		end
+
+		def configure(conf)
+			super
+			@log = nil
+			
+			if @enable_log
+				@log = Logger.new(@log_path, 'weekly')
+				@log.debug {'Starting filter_inventory2mdm plugin'}
+			end
+		end
+
+        def start
+            super
+            @process_incoming_stream = check_custom_metrics_availability
+            @log.debug "After check_custom_metrics_availability process_incoming_stream #{@process_incoming_stream}"
+        end
+
+        def check_custom_metrics_availability
```

Comment:
This function is in multiple places now. Good reason to put it as a library & ref it..